web/checks_api.go

```diff
@@ -28,24 +27,76 @@ type JSONCheck struct {
 	Labels         string `json:"labels,omitempty"`
 }
 
-// ApiCheckResultsHandler godoc
-// @Summary Get a specific cluster's check results
+type JSONChecksResults struct {
+	Hosts  map[string]*JSONHosts       `json:"hosts,omitempty" binding:"required"`
+	Checks map[string]*JSONCheckResult `json:"checks,omitempty" binding:"required"`
+}
+
+type JSONHosts struct {
+	Result    string `json:"result,omitempty"`
+	Reachable bool   `json:"reachable,omitempty"`
+	Msg       string `json:"msg,omitempty"`
+}
+
+type JSONCheckResult struct {
+	ID          string                `json:"id,omitempty"`
+	Hosts       map[string]*JSONHosts `json:"hosts,omitempty"`
+	Group       string                `json:"group,omitempty"`
+	Description string                `json:"description,omitempty"`
+}
+
+// ApiGetCheckResultsHandler godoc
+// @Summary Get a specific group's check results
 // @Produce json
-// @Param cluster_id path string true "Cluster Id"
-// @Success 200 {object} map[string]interface{}
+// @Param id path string true "Resource Id"
+// @Success 200 {object} JSONChecksResults
+// @Failure 404 {object} map[string]string
+// @Router /api/checks/{id}/results [get]
+func ApiGetCheckResultsHandler(s services.ChecksService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		id := c.Param("id")
+
+		checkResults, err := s.GetChecksResultAndMetadataById(id)
+		if err != nil {
+			c.Error(NotFoundError("could not find checks group"))
+			return
+		}
+
+		c.JSON(http.StatusOK, checkResults)
+	}
+}
+
+// ApiCreateChecksResultstaHandler godoc
+// @Summary Create a checks results entry
+// @Produce json
+// @Param id path string true "Resource Id"
+// @Param Body body JSONChecksResults true "Checks results"
+// @Success 201 {object} JSONChecksResults
 // @Failure 500 {object} map[string]string
-// @Router /api/clusters/{cluster_id}/results [get]
-func ApiClusterCheckResultsHandler(client consul.Client, s services.ChecksService) gin.HandlerFunc {
+// @Router /api/checks/{id}/results [post]
+func ApiCreateChecksResultstaHandler(s services.ChecksService) gin.HandlerFunc {
```

Comment:
The name of the function seems to have a typo: `ApiCreateChecksResultstaHandler` -> `ApiCreateChecksResultsHandler`